api-tests/src/test/java/org/openmrs/module/reporting/report/definition/service/ReportDefinitionServiceImplTest.java

```diff
@@ -43,7 +43,24 @@ public void purgeDefinition_shouldPurgeReportDesigns() throws Exception {
     	ReportDefinition definition = reportDefinitionService.getDefinition(80);
     	reportDefinitionService.purgeDefinition(definition);
 	    
-    	ReportDesign reportDesign = reportService.getReportDesign(2);
-    	assertThat(reportDesign, nullValue());
+		assertThat(reportService.getReportDesign(3), nullValue());
+		assertThat(reportService.getReportDesign(4), nullValue());
+	}
+
+	/**
+	 * @see ReportDefinitionServiceImpl#purgeDefinition(String)
+	 * @verifies purge report definition by uuid and associated report designs and requests
+	 */
+	@Test
+	public void purgeDefinition_shouldPurgeDesignsAndRequestsAndDefinition() throws Exception {
+		executeDataSet("org/openmrs/module/reporting/include/ReportDefinitionServiceImplTest.xml");
+
+		reportDefinitionService.purgeDefinition("c11f5354-9567-4cc5-b3ef-163e28873926");
+
+		assertNull(reportService.getReportDesign(3));

```

Comment:
Good point! I tried to add some guard assertions, but got a failure in the assertNull tests, they are all failing now. Without the extra assertions, they pass. I'm not sure yet what's causing it, need to investigate more.